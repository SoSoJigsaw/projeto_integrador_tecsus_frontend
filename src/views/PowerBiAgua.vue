<template>
  <transition name="fade" mode="out-in">
    <div v-if="isVisible" class="container-powerbi">
      <h1>Dashboard de √Ågua</h1>
      <iframe :src="powerBiSrc" frameborder="0" allowfullscreen="true"></iframe>
    </div>
  </transition>
</template>

<script lang="ts">
export default {
  data() {
    return {
      isVisible: false,
      powerBiSrc: 'https://app.powerbi.com/groups/me/reports/aa6a9cb1-9104-489a-9214-23cbb5bf6e5a?ctid=cf72e2bd-7a2b-4783-bdeb-39d57b07f76f&pbi_source=linkShare&bookmarkGuid=69166d83-b5c7-4210-8793-103aeae51f77',
    }
  },

  mounted() {
    setTimeout(() => {
      this.isVisible = true
    }, 1000)
  }
}
</script>

<style>
@import '../assets/base.css';

.container-powerbi {
  margin-left: 4%;
  margin-top: 3%;
  margin-right: 4%;
  margin-bottom: 3%;
  display: flex;
  flex-direction: column;
  gap: 4.5rem;
}

.container-powerbi h1 {
  color: #776ae3;
  font-size: 6vmin;
  font-weight: 400;
}

.container-powerbi iframe {
  align-self: center;
  justify-self: center;
}
</style>
