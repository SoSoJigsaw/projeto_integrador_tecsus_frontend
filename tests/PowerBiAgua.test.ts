import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import PowerBiAgua from '@/views/PowerBiAgua.vue'

it('should mount the component correctly', () => {
  const wrapper = mount(PowerBiAgua)
  expect(wrapper.exists()).toBe(true)
})

it('should start with isVisible set to false', () => {
  const wrapper = mount(PowerBiAgua)
  expect(wrapper.vm.isVisible).toBe(false)
})

it('should display the main container after 1 second', async () => {
  vi.useFakeTimers()
  const wrapper = mount(PowerBiAgua)
  vi.advanceTimersByTime(1000)
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-powerbi').isVisible()).toBe(true)
  vi.useRealTimers()
})

it('should render iframe with the correct Power BI source', () => {
  const wrapper = mount(PowerBiAgua)
  const iframe = wrapper.find('iframe')
  expect(iframe.exists()).toBe(true)
  expect(iframe.attributes('src')).toBe(wrapper.vm.powerBiSrc)
})

it('should display the home and light dashboard icons', () => {
  const wrapper = mount(PowerBiAgua)
  expect(wrapper.find('.fa-house').exists()).toBe(true)
  expect(wrapper.find('img[src="../assets/icons/luz.png"]').exists()).toBe(true)
})

it('should increase icon opacity on hover', async () => {
  const wrapper = mount(PowerBiAgua)
  const img = wrapper.find('img[src="../assets/icons/luz.png"]')
  await img.trigger('mouseover')
  expect(img.element.style.opacity).toBe('1')
})

it('should navigate to home when home icon is clicked', async () => {
  const pushSpy = vi.fn()
  const wrapper = mount(PowerBiAgua, {
    global: {
      mocks: {
        $router: {
          push: pushSpy
        }
      }
    }
  })
  await wrapper.find('.fa-house').trigger('click')
  expect(pushSpy).toHaveBeenCalledWith('/')
})

it('should display the water icon when not visible', () => {
  const wrapper = mount(PowerBiAgua)
  expect(wrapper.find('.entre-paginas img[src="../assets/icons/agua.png"]').exists()).toBe(true)
})

it('should react to a manual change in isVisible', async () => {
  const wrapper = mount(PowerBiAgua)
  wrapper.vm.isVisible = true
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-powerbi').isVisible()).toBe(true)
})

it('should render without errors with default props', () => {
  const wrapper = mount(PowerBiAgua)
  expect(wrapper.html()).toMatchSnapshot()
})
