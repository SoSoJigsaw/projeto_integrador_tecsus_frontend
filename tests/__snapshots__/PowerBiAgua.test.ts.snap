// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render without errors with default props 1`] = `
"<transition-stub name="fade" mode="out-in" appear="false" persisted="false" css="true">
  <div class="entre-paginas"><img src="/src/assets/icons/agua.png"></div>
</transition-stub>"
`;
